### ISSUE TITLE
Browser support?

### ISSUE THREAD

**Issue by CharlesStover** [mysticatea/abort-controller#8]
I can't find anywhere that documents what the browser support for this shim is.

How does it compare to the innate `AbortController` support? How much market share, if any, am I gaining by using this?

**Comment by CharlesStover** [comment]
@mysticatea

**Comment by mysticatea** [comment]
I'm sorry for the overlook.

You can use this package on browsers normally. Just import it by a script element or use Webpack or something like.

`AbortController` provides the feature that notifies `abort` events to subscribers. It's all. This shim doesn't touch what the subscribers of `abort` event do.

**Comment by CharlesStover** [comment]
@mysticatea  So you're saying that if I fetch something in Internet Explorer 6, this AbortController will be able to abort that request? I am asking what browsers this gives functionality to that don't already support AbortController. I know that abort is not supported by many browsers, so I am wondering how many browsers this is capable of giving abort functionality to. Or maybe I am misunderstanding the purpose of the repo.

**Comment by mysticatea** [comment]
No. This is an implementation of `AbortController` API, but this is not a polyfill of `fetch`. The `fetch` API accepts `signal` option and subscribes the `abort` event, but it's a feature of the `fetch` API.

**Comment by CharlesStover** [comment]
@mysticatea So right now, I am able to use `AbortController` as built into Chrome. Why import an implementation of `AbortController` when the browser already supports it?

I presume the reason is to polyfill browsers that support abort but not AbortController.

**Comment by mysticatea** [comment]
AbortController is an independent API for all features which need to abort ongoing activities. In the spec, [Aborting ongoing activities](https://dom.spec.whatwg.org/#aborting-ongoing-activities) section mentions it.

Originately, people have need the standard way to cancel asyncronous jobs, and it had discussed as cancellable promises in TC39. However, the cancellable promises spec had been withdrown. So people defined the standard way in WHATWG.

Fetch API, WebRTC, WebUSB, ....
I think there are several APIs need the way which aborts asyncronous jobs. It's convenient if the way is common in all Web APIs. And WHATWG mentions that AbortController is the common way.

I expect the polyfills of the APIs which depend on AbortController to use this package rather than each polyfill implements it.

**Comment by CharlesStover** [comment]
@mysticatea Maybe this is more clear:
Do you know which browser versions support signals on fetch (`fetch(..., { signal: ...})`) but _do not_ support `AbortController`?

This is essentially a polyfill for `AbortController`, right? If `AbortController` isn't defined in your browser, this defines it. `AbortController` [comes built into Chrome 66, Edge 16, Firefox 57, Opera 43, and Safari 11.1](https://developer.mozilla.org/en-US/docs/Web/API/AbortController#Browser_compatibility).
If I am using one of the aforementioned browsers that innately supports `AbortController`, this package is unnecessary, right?
This package is for browsers that do not have support for `AbortController` already built in.

My question is, if a browser does **not** innately support `AbortController`, _why define it_?
I am under the impression that a browser that _does not innately support_ `AbortController` **also does not** support the `signal` option on `fetch`.

What does a polyfill of `AbortController` _do_ if it can't abort a fetch request?
Are there browsers that _can_ abort a fetch request but _do not_ have `AbortController` built in?

**Comment by mysticatea** [comment]
> My question is, if a browser does not innately support AbortController, why define it?

To implement a polyfill of the feature which depends on AbortController.

**Comment by mysticatea** [comment]
For example, if you want to implement a polyfill of the whole `fetch` API, you can use this package, then it may reduce duplicate code with the polyfills of other APIs which depend on `AbortController` (`AbortController` is a standard API for aborting asynchronous jobs in WHATWG layer). Especially, if the polyfill aims at isomorphic, this package would be useful since this works on Node.js as well.

Of cause, if you are interested in only modern browsers, you don't need this package.

**Comment by CharlesStover** [comment]
@mysticatea

I am under the impression that any browser that supports the `signal` option of `fetch` already supports `AbortController` without a polyfill. And that any browser that does not have `AbortController` innately available _does not_ support the `signal` option of `fetch`.

That is to say that adding an `AbortController` polyfill _does nothing_ if the browser does not support `signal` on `fetch` in the first place. For example, Chrome 65 does not have innate `AbortController` support. Using this polyfill _does not_ allow `fetch` to be aborted, because the `signal` option of `fetch` is completely ignored.

Why polyfill it if it does nothing?
Are there browsers that do have `signal` support on `fetch` but do not have `AbortController` innately available?

Again, for example, I used this polyfill on Chrome 65, and _it did nothing_, because Chrome 65 does not support `signal` on `fetch`.
On Chrome 66, where `signal` is a supported option of `fetch`, `AbortController` is already defined innately, and so this polyfill still does nothing.

**Comment by mysticatea** [comment]
I'm not sure what this issue's goal.

This is not the polyfill of the `fetch` API. This is an isomorphic implementation of `AbortController` API. The `fetch` API is merely one of APIs which depend on `AbortController` API. The `AbortController` API does not depends on `fetch` API. The dependency is not bidirection.
